### PR TITLE
Add the testing readme

### PR DESCRIPTION
**Pull request by fedepaol**
# Description

This PR adds a readme to describe the status of the unit / e2e tests, how to contribute to them and how to integrate them into the OCP CI.

Fixes # (issue)

## Type of change

Please select the appropiate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [x] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: features/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+The name of the test entry point must reflect the name of the feature folder under [features](./features). This will make adding new jobs easier.
+
+Each test must not perform any kind of setup, and test the feature assuming that it's already avaliable in the cluster we are testing against.
+
+## Running the e2e tests for a given set of features
+
+This can be done by running a command like
+
+```bash
+FEATURES="sctp ptp" && make functests
```
Comment (fromanirh) [inline features/TESTING.md]: depending on what you want to do, the "&&" is redundant here

File: features/TESTING.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+## Deployment
+
+In order to ease the creation of e2e test jobs, each feature should provide a `deploy.sh` script that will apply the given feature to the cluster and will wait for it to be available (i.e. waiting for machine configs to be applied).
+
+If this convention is followed, applying a given set of features will be driven by the same `FEATURES` variable:
+
+```bash
+FEATURES="sctp ptp" && make deploy
```
Comment (fromanirh) [inline features/TESTING.md]: same

File: features/TESTING.md (hunk 1 above)
Comment (fedepaol) [reply features/TESTING.md]: Ah, nice. Thanks!

**Comment by slintes** [comment]
:+1: 
/lgtm

**Pull request merged**